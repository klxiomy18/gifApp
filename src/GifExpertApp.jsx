import { useState } from "react"
import { AddCategory, GifGrid } from "./components"


export const GifExpertApp = () => {

    const [categories, setCategories] = useState([''])

    const onAddCategory = (newCategory) => { 

        if(categories.includes(newCategory)) return;

        setCategories([newCategory, ...categories])
    }
    
    
  return (
    <>
        {/* titulo */}
        <h1>GifExpertApp</h1>

        {/* input  */}
        <AddCategory
        
        onNewCategory={ onAddCategory }
         />

        {/* Listado de gifs  */}
        {/* <button onClick={onAddCategory} >Agregar</button> */}
        
            { 
            categories.map( category => (
                <GifGrid category={category} key={category}/>
            ))
            }

    </>
  )
}

