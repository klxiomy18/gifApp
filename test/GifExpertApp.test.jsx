import { render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"


describe('Pruebas en GifExpertApp', () => {
  
    test('Debe hacer match con el snapshot', () => {
      const {container} = render(<GifExpertApp/>);
    //   screen.debug()
        expect(container).toMatchSnapshot()
    });

    test('Debe aparecer el titulo, value y cargando', () => {
      const title = 'GifExpertApp';
      const value = 'One punch';

      render(<GifExpertApp/>);
      const InputValue = screen.getByRole('textbox');
      
      expect(screen.getByText(title));
      expect(InputValue.value).toBe(value);
      expect(screen.getByText('Cargando...'))
    })
    
    
})
