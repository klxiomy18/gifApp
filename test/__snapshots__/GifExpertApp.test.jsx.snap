// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    action=""
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value="One punch"
    />
  </form>
  <h3 />
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
